tfhe/src/core_crypto/algorithms/lwe_programmable_bootstrapping/fft64.rs

```diff
@@ -1071,3 +1071,77 @@ pub fn programmable_bootstrap_lwe_ciphertext_mem_optimized_requirement<OutputSca
 ) -> Result<StackReq, SizeOverflow> {
     bootstrap_scratch::<OutputScalar>(glwe_size, polynomial_size, fft)
 }
+
+/// Memory optimized version of batch_programmable_bootstrap_lwe_ciphertext, the caller must
+/// provide a properly configured [`FftView`] object and a `PodStack` used as a memory buffer having
+/// a capacity at least as large as the result of
+/// [`crate::core_crypto::algorithms::batch_programmable_bootstrap_lwe_ciphertext_mem_optimized_requirement`].
+pub fn batch_programmable_bootstrap_lwe_ciphertext_mem_optimized<
+    InputScalar,
+    OutputScalar,
+    InputCont,
+    OutputCont,
+    AccCont,
+    KeyCont,
+>(
+    input: &LweCiphertextList<InputCont>,
+    output: &mut LweCiphertextList<OutputCont>,
+    accumulator: &GlweCiphertextList<AccCont>,
+    fourier_bsk: &FourierLweBootstrapKey<KeyCont>,
+    fft: FftView<'_>,
+    stack: PodStack<'_>,
+) where
+    // CastInto required for PBS modulus switch which returns a usize
+    InputScalar: UnsignedTorus + CastInto<usize>,
+    OutputScalar: UnsignedTorus,
+    InputCont: Container<Element = InputScalar>,
+    OutputCont: ContainerMut<Element = OutputScalar>,
+    AccCont: Container<Element = OutputScalar>,
+    KeyCont: Container<Element = c64>,
+{
+    assert_eq!(
+        accumulator.ciphertext_modulus(),
+        output.ciphertext_modulus(),
+        "Mismatched moduli between accumulator ({:?}) and output ({:?})",
+        accumulator.ciphertext_modulus(),
+        output.ciphertext_modulus()
+    );
+
+    assert_eq!(
+        fourier_bsk.input_lwe_dimension(),
+        input.lwe_size().to_lwe_dimension(),
+        "Mismatched input LweDimension. \
+        FourierLweBootstrapKey input LweDimension: {:?}, input LweCiphertext LweDimension {:?}.",
+        fourier_bsk.input_lwe_dimension(),
+        input.lwe_size().to_lwe_dimension(),
+    );
+    assert_eq!(
+        fourier_bsk.output_lwe_dimension(),
+        output.lwe_size().to_lwe_dimension(),
+        "Mismatched output LweDimension. \
+        FourierLweBootstrapKey input LweDimension: {:?}, input LweCiphertext LweDimension {:?}.",
+        fourier_bsk.output_lwe_dimension(),
+        output.lwe_size().to_lwe_dimension(),
+    );
+
+    fourier_bsk.as_view().batch_bootstrap(
+        output.as_mut_view(),
+        input.as_view(),
+        &accumulator.as_view(),
+        fft,
+        stack,
+    );
+}
+
+/// Return the required memory for [`batch_programmable_bootstrap_lwe_ciphertext_mem_optimized`].
+pub fn batch_programmable_bootstrap_lwe_ciphertext_mem_optimized_requirement<OutputScalar>(
+    glwe_size: GlweSize,
+    polynomial_size: PolynomialSize,
+    fft: FftView<'_>,
+) -> Result<StackReq, SizeOverflow> {
+    programmable_bootstrap_lwe_ciphertext_mem_optimized_requirement::<OutputScalar>(
+        glwe_size,
+        polynomial_size,
+        fft,
+    )
```

Comment:
actually reading the code this needs to be updated as we copy several GLWEs, so we need to copy the code of the pbs mem requirement, and multiply the scratch space for GLWEs by the number of LUTs, we can discuss taht over a call if you want